design/authenticators/authn_gcp/authn_gcp_solution_design.md

```diff
@@ -0,0 +1,573 @@
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-gcp/<service ID>`
+- !policy
+  id: conjur/authn-gcp/mygcp
+  body:
+  - !webservice
+      
+  - !group apps
+  
+  - !permit
+    role: !group apps
+    privilege: [ read, authenticate ]
+    resource: !webservice
+```
+
+All the following configurations will have constants values and will not require any action from the customer:
+
+* **provider-uri** = `https://www.googleapis.com/oauth2/v1/certs` #represnts the address to fetch the public key in order to validate the JWT token
+* **mandatory-claims** = `iss, exp, aud, iat` #the list of the mandatory claims to validate in the JWT token
+* **mandatory-claims** = `iss, exp, aud, iat, nbf` #the list of the mandatory claims to validate in the JWT token
+* **iss** = `https://accounts.google.com` #the issuer claim we should validate in the JWT token
+* **aud** = `conjur` #the audience claim we should validate in the JWT token
+
+### GCP Resource Restrictions
+To authenticate with Conjur, GCP-specific fields will need to be provided in the host annotations of the Conjur host identity with the following options:
+
+* **project_id** - A customizable unique identifier for your project.
+* **instance_name** - The name of the GCE machine
+* **service_account_id** - The google instance identity   
+* **service_account_email** - The email of the instance identity
+
+At least one of the above should be provided, if more than 1 is provided, we will authenticate them with AND logic.
+
+A Conjur host will be defined as follows with their annotations holding GCP-specific identification properties.
+
+```yaml
+- !policy
+  id: <policy-id>
+  body:
+    - !group
+ 
+    - &hosts
+      - !host
+        id: myapp
+        annotations:
+          authn-gcp/instance-name: <instance-name>
+          authn-gcp/project-id: <project-id>
+          authn-gcp/service-account-id: <service-account-id>
+          authn-gcp/service-account-email: <service-account-email>
+           
+    - !grant
+      role: !group
+      members: *hosts
+          
+- !grant
+  role: !group /conjur/authn-gcp/mygcp/apps
+  member: !group <policy-id>
+```
+
+### Environment-provided service account google identity
+In order to authenticate against Conjur GCP authenticator, a GCE instance first need to fetch its environment-provided identity.
+
+**Example:** 
+Run the following from the GCE machine:
+
+`curl \ --header Metadata-Flavor: Google" \ --get \ --data-urlencode "audience=conjur" \ --data-urlencode "format=full" \ "http://metadata/computeMetadata/v1/instance/service-accounts/default/identity`
+
+The audience value must to be `conjur` 
+
+JWT token:
+
+```json
+{
+  "aud": "http://vault/conjur",
+  "azp": "110987294251917851298",
+  "email": "716149158341-compute@developer.gserviceaccount.com",
+  "email_verified": true,
```

Comment:
@InbalZilberman @nessiLahav 
Is it something that we need to be aware on?
In which cases is it `false`?

Reply:
good point i will add a card to investigate it